### PR TITLE
feat(af-core): FORMS-25463 expose cq:annotations parallel to fd:dor

### PR DESCRIPTION
**Pull request by prateek1497**
Adds cq:annotations read support to the IC GET flow for the print channel /
   author CRISPR path.

  - getCqAnnotations() reads the cq:annotations child resource (a sibling of fd:dorContainer on the component node) and returns each annotation as a map entry keyed by node name. Each entry carries color, text, x, y, and optional state/resolvedBy/resolvedAt plus JCR audit fields. Null fields are stripped so the serialized shape matches the JCR content exactly.
  - getProperties() is wired to call getCqAnnotations() and place the result under properties.cq:annotations — parallel to fd:dor / fd:path / fd:associate. Returns null when no cq:annotations child exists; Jackson omits the key, keeping output non-breaking for forms without annotations.
  - getCustomProperties() explicitly excludes cq:annotations to prevent it leaking as a double-serialized JSON string via the catch-all custom-properties path.
  - Gated: only runs when channel == PRINT and isAuthorMode(request) — web channel and publish mode are unaffected.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes and the overall coverage did not decrease.
- [ ] All unit tests pass on CircleCi.
- [ ] I ran all tests locally and they pass.

### CODE REVIEW

**Comment by prateek1497** [comment]
In short why its not following AFv2/sites annotations - because it does network call for each annotation, bringing the infinity json for that annotation, for IC, we would add cq:annotations in the first GET call, that is adding it in authoring crispr, and flushing everything on the UI at once. and same in the existing SAVE call of Delrina, all annotation would get save at once. We have made changes at CoreComponentFormGenerator for this

File: bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/util/AbstractFormComponentImpl.java [hunk 1]
```diff
@@ -740,4 +747,31 @@ public Map<String, Object> getDorContainer() {
...
+        }
+        Resource annotationsResource = resource.getChild("cq:annotations");
+        if (annotationsResource == null) {
+            return null;
+        }
+        Map<String, Object> result = new LinkedHashMap<>();
+        for (Resource child : annotationsResource.getChildren()) {
+            ValueMap vm = child.getValueMap();
+            Map<String, Object> ann = vm.entrySet().stream()
+                .filter(e -> isAllowedType(e.getValue()))
```
Comment (rismehta) [inline bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/util/AbstractFormComponentImpl.java]: **Warning — JCR internal properties leak into annotation output**

`isAllowedType` accepts `String`, so `jcr:primaryType = "nt:unstructured"` passes the filter. Every annotation node will have this key in its exported map. Other JCR audit fields (`jcr:created`, `jcr:createdBy`, `jcr:modified`, etc.) stored as `Calendar`/`String` will also leak.

Mirror the prefix exclusion from `getCustomProperties()`:
```java
.filter(e -> isAllowedType(e.getValue())
    && !e.getKey().startsWith("jcr:")
    && !e.getKey().startsWith("sling:"))
```
Also add a test assertion: `assertNull(annotation.get("jcr:primaryType"))`.

File: bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/util/AbstractFormComponentImpl.java [hunk 2]
```diff
@@ -643,6 +649,7 @@ private Map<String, Object> getCustomProperties() {
             .stream()
             .filter(entry -> isAllowedType(entry.getValue())
                 && !reservedProperties.contains(entry.getKey())
+                && !CUSTOM_ANNOTATIONS_PROPERTY_WRAPPER.equals(entry.getKey())
```
Comment (rismehta) [inline bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/util/AbstractFormComponentImpl.java]: **Warning — `ReservedProperties` convention bypassed**

Per `docs/architecture/overview.md` §8.2, every JCR property name must be registered in `ReservedProperties.java` — `aggregateReservedProperties()` auto-collects all `String` fields. The PR adds a one-off inline guard instead.

Add to `ReservedProperties.java`:
```java
public static final String PN_CQ_ANNOTATIONS = "cq:annotations";
```
Then the inline filter here becomes unnecessary (once the constant is in `ReservedProperties`, `reservedProperties.contains("cq:annotations")` is true and the existing guard covers it).

File: bundles/af-core/src/test/java/com/adobe/cq/forms/core/components/util/AbstractFormComponentImplTest.java [hunk 3]
```diff
@@ -273,6 +274,57 @@ public void testPrintChannelRule() {
...
+        Map<String, Object> properties = abstractFormComponentImpl.getProperties();
+        Map<String, Object> annotations = (Map<String, Object>) properties.get("cq:annotations");
+        assertNotNull(annotations, "cq:annotations should appear for print channel");
+        Map<String, Object> annotation = (Map<String, Object>) annotations.get("annotation-1");
+        assertNotNull(annotation);
+        assertEquals("review note", annotation.get("text"));
+    }
+
+    @Test
+    public void testAnnotationPropertiesIncludeAllAllowedTypes() {
```
Comment (rismehta) [inline bundles/af-core/src/test/java/com/adobe/cq/forms/core/components/util/AbstractFormComponentImplTest.java]: **Warning — test name misrepresents coverage**

`testAnnotationPropertiesIncludeAllAllowedTypes` only exercises `String` and `Long`. `isAllowedType` also accepts `Boolean`, `Calendar`, `BigDecimal` and their array variants — none are covered. Rename to `testAnnotationPropertiesIncludeStringAndLongTypes`, or add fixture data and assertions for at least one additional type.

**Review by rismehta — COMMENTED** [review]
## Code Review — PR #1883: expose cq:annotations parallel to fd:dor

### Summary

Solid feature addition with good channel-scoping logic and defensive null checks. Two **Warnings** need fixing before merge, plus a few suggestions.

---

### Positive
- Good early-return guards (null resource, non-print channel).
- Correct Mockito stub added to `testAssociateProperties` — necessary because STRICT_STUBS mode causes failure if `resource.getChild("cq:annotations")` is invoked without a stub.
- `LinkedHashMap` preserves annotation node order.

---

### Suggestions
- **`getCqAnnotations()` visibility**: method is `public` with no interface declaration. Analogous `getAssociateProperties()` is `private`. Change to `private` unless a subclass override use case is documented.
- **Missing edge-case test**: no test covers the case where a `cq:annotations` child node exists but has zero annotation children (method returns null — branch untested).
- **`reviewedBy`/`reviewedAt` in fixture**: these fields appear in the test fixture but are not listed in the PR description schema. Either assert them or clarify.
- `if (annotations != null && !annotations.isEmpty())` in `getProperties()` is redundant since `getCqAnnotations()` already returns null for empty — minor cleanup.

File: bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/util/AbstractFormComponentImpl.java (hunk 1 above)
Comment (prateek1497) [reply bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/util/AbstractFormComponentImpl.java]: added this

File: bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/util/AbstractFormComponentImpl.java (hunk 2 above)
Comment (prateek1497) [reply bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/util/AbstractFormComponentImpl.java]: added this

File: bundles/af-core/src/test/java/com/adobe/cq/forms/core/components/util/AbstractFormComponentImplTest.java (hunk 3 above)
Comment (prateek1497) [reply bundles/af-core/src/test/java/com/adobe/cq/forms/core/components/util/AbstractFormComponentImplTest.java]: fixed this

**Review by rismehta — APPROVED** [review]
